test/search_test.dart

```diff
@@ -264,4 +265,29 @@ void main() {
       });
     });
   });
+
+  group('with finite-pagination query params', () {
+    test('with basic query', () async {
+      var index = await createBooksIndex();
+      var result = await index.search('pri', page: 1) as PaginatedSearchResult;
+
+      expect(result.totalPages, 1);
+    });
+
+    test('accesses fields from Searchable', () async {
+      var index = await createBooksIndex();
+      var result = await index.search('pri', page: 1) as PaginatedSearchResult;
+
+      expect(result.hits!.length, greaterThanOrEqualTo(1));
+      expect(result.totalPages, 1);
+    });
```

Comment:
I'm not sure to understand the differences between these two tests.
It called `accesses fields from Searchable` but it didn't really access to hits, no?

Reply:
The first one is just a simple query passing the parameters. And the second is to exemplify how you can access both types of fields (from the abstract class) and the implementation class.